api/desecapi/management/commands/scavenge-unused.py

```diff
@@ -0,0 +1,109 @@
+import datetime
+from functools import reduce
+import operator
+
+from django.conf import settings
+from django.core.mail import get_connection, mail_admins
+from django.core.management import BaseCommand
+from django.db import transaction
+from django.db.models import F, Q
+from django.test import RequestFactory
+from django.utils import timezone
+from rest_framework.reverse import reverse
+
+from desecapi import models, serializers, views
+from desecapi.pdns_change_tracker import PDNSChangeTracker
+
+
+inactive_days = 183
+notice_days_fresh = 28
+notice_days_notified = 7
+
+
+class Command(BaseCommand):
+    lps_children = models.Domain.objects.filter(reduce(
+        operator.or_,
+        (Q(name__endswith=f'.{lps}') for lps in settings.LOCAL_PUBLIC_SUFFIXES),
+        Q(pk__in=[])  # always-false default
+    ))
+
+    @classmethod
+    def renew_touched_domains(cls):
+        expiry_candidates = cls.lps_children.exclude(renewal_state=models.Domain.RenewalState.FRESH)
+
+        last_published_threshold = timezone.localdate() - datetime.timedelta(days=183)
+        recently_touched_domains = expiry_candidates.filter(published__date__gte=last_published_threshold)
+
+        print('Renewing domains:', *recently_touched_domains.values_list('name', flat=True))
+        recently_touched_domains.update(renewal_state=models.Domain.RenewalState.FRESH, renewal_changed=F('published'))
+
+    @classmethod
+    def warn_domain_deletion(cls, renewal_state, notice_days, inactive_days):
```

Comment:
maybe add some documentation to this method

Reply:
done (method also simplified meanwhile, as `renewal_changed` no longer can be `None`)